src/core/shape_inference/include/ctc_greedy_decoder_seq_len_shape_inference.hpp

```diff
@@ -0,0 +1,57 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+#pragma once
+#include <openvino/op/ctc_greedy_decoder_seq_len.hpp>
+
+namespace ov {
+namespace op {
+namespace v6 {
+
+template <class T>
+void shape_infer(const CTCGreedyDecoderSeqLen* op, const std::vector<T>& input_shapes, std::vector<T>& output_shapes) {
+    NODE_VALIDATION_CHECK(op, (input_shapes.size() == 2 || input_shapes.size() == 3) && output_shapes.size() == 2);
+    using DimType = typename std::iterator_traits<typename T::iterator>::value_type;
+    const auto& logits_shape = input_shapes[0];
+    const auto& seq_len_shape = input_shapes[1];
+    const bool logits_is_static_rank = logits_shape.rank().is_static();
+    const bool seq_len_is_static_rank = seq_len_shape.rank().is_static();
+    auto& decoded_shape = output_shapes[0];
+    auto& seq_shape = output_shapes[1];
+    decoded_shape.resize(2);
+    seq_shape.resize(1);
+    if (input_shapes.size() == 3) {
+        const auto& blank_shape = input_shapes[2];
+        const auto& blank_rank = blank_shape.rank();
+        if (blank_rank.is_static() && blank_shape.is_static()) {
```

Comment:
could be checked via blank_shape.is_static() only

Reply:
Great ,will change it